micromouse/RobotIO.cpp

```diff
@@ -552,4 +553,107 @@ namespace Micromouse
 		IRSensors[FRONT_LEFT]->saveCalibration(IR_FRONT_LEFT_MEMORY);
 		IRSensors[FRONT_RIGHT]->saveCalibration(IR_FRONT_RIGHT_MEMORY);
 	}
+
+
+
+	bool RobotIO::checkMotors()
+	{
+		int startCountsRight = 0;
+		int startCountsLeft = 0;
+		int finishCountsRight = 0;
+		int finishCountsLeft = 0;
+		int encoderRight = 0;
+		int encoderLeft = 0;
+
+		//Zero encoder value and sets encoder counts to temp var-
+		//reads value before -
+		//runs motor for 2 turns and 
+
+		startCountsLeft = leftMotor.getCounts();
+		startCountsRight = rightMotor.getCounts();
+
+		leftMotor.setMovement(0.5f);
+		rightMotor.setMovement(0.5f);
+
+
+#ifdef __MK20DX256__ // Teensy Compile
+		delay(1000);
+#endif		
+
+		leftMotor.brake();
+		rightMotor.brake();
+
+		encoderLeft = leftMotor.getCounts();
+		encoderRight = rightMotor.getCounts();
+
+
+		if (finishCountsLeft > 0 && finishCountsRight > 0) {
```

Comment:
0 is not a good number to check against. when the encoders are not working the value goes between -1 and 1.
the motors were on half speed for 1 second, safe to assume over 1000 counts